packages/babel-plugin-jest-hoist/src/index.ts

```diff
@@ -183,6 +183,12 @@ export default (): {visitor: Visitor} => {
         path.node._blockHoist = Infinity;
       }
     },
+    ImportDeclaration(path) {
```

Comment:
Should probably look for a `require` instead (or as well?)

Reply:
I tried this:

```ts
const visitor: Visitor = {
  CallExpression(path) {
    const callee = path.get('callee') as NodePath;
    const callArguments = path.get('arguments');

    if (
      callee.isIdentifier() &&
      callee.node.name === 'require' &&
      callArguments.length === 1
    ) {
      const [argument] = callArguments;

      if (
        argument.isStringLiteral() &&
        argument.node.value === '@jest/globals'
      ) {
        // @ts-ignore: private, magical property
        path.node._blockHoist = Infinity;
      }
    }
  },
};
```

and it gets into the `_blockHoist` thing, but doesn't seem to actually hoist it. Halp? 😀

Reply:
New try:

```ts
const visitor: Visitor = {
VariableDeclaration(path) {
    const declarations = path.get('declarations');
  
    if (declarations.length === 1) {
      const declarationInit = declarations[0].get('init');
  
      if (declarationInit.isCallExpression()) {
        const callee = declarationInit.get('callee') as NodePath;
        const callArguments = declarationInit.get('arguments') as Array<
          NodePath
        >;
  
        if (
          callee.isIdentifier() &&
          callee.node.name === 'require' &&
          callArguments.length === 1
        ) {
          const [argument] = callArguments;
  
          if (
            argument.isStringLiteral() &&
            argument.node.value === '@jest/globals'
          ) {
            // @ts-ignore: private, magical property
            path.node._blockHoist = Infinity;
          }
        }
      }
    }
  }
}
```

This seems to hoist the `require`, but then the `jest.mock` call is _not_ hoisted

```js
"use strict";

var _globals = require("@jest/globals");

/**
 * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */
// The virtual mock call below will be hoisted above this `require` call.
var virtualModule = require('virtual-module');

_globals.jest.mock('virtual-module', function () {
  return 'kiwi';
}, {
  virtual: true
});

test('works with virtual modules', function () {
  expect(virtualModule).toBe('kiwi');
});
```

Reply:
I pushed up that last one, since it feels like it's somewhere close